includes/popups-analytics/class-popups-analytics-utils.php

```diff
@@ -0,0 +1,281 @@
+	 */
+	public static function get_ga_report( $options ) {
+		$offset = $options['offset'];
+
+		// Load and query analytics.
+		if ( defined( 'GOOGLESITEKIT_PLUGIN_MAIN_FILE' ) ) {
+			$context   = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+			$analytics = new Analytics( $context );
+
+			if ( $analytics->is_connected() ) {
+				// Create the DateRange object.
+				$date_range = new Google_Service_AnalyticsReporting_DateRange();
+				$start_date = gmdate( 'Y-m-d', strtotime( $offset . ' days ago' ) );
+				$end_date   = gmdate( 'Y-m-d', strtotime( '1 days ago' ) );
+				$date_range->setStartDate( $start_date );
+				$date_range->setEndDate( $end_date );
+
+				// Create the Metrics object.
+				$metrics = new Google_Service_AnalyticsReporting_Metric();
+				$metrics->setExpression( 'ga:totalEvents' );
+				$metrics->setAlias( 'events' );
+
+				// Filter just the popups custom event category.
+				$dimension_category_filter = new Google_Service_AnalyticsReporting_SegmentDimensionFilter();
+				$dimension_category_filter->setDimensionName( 'ga:eventCategory' );
+				$dimension_category_filter->setOperator( 'EXACT' );
+				$dimension_category_filter->setExpressions( array( self::EVENT_CATEGORY ) );
+
+				// Create the DimensionFilterClauses.
+				$dimension_filter_clause = new Google_Service_AnalyticsReporting_DimensionFilterClause();
+				$dimension_filter_clause->setFilters( array( $dimension_category_filter ) );
+
+				// Create the Dimension objects.
+				$date_dimension = new Google_Service_AnalyticsReporting_Dimension();
+				$date_dimension->setName( 'ga:date' );
+				$action_dimension = new Google_Service_AnalyticsReporting_Dimension();
+				$action_dimension->setName( 'ga:eventAction' );
+				$label_dimension = new Google_Service_AnalyticsReporting_Dimension();
+				$label_dimension->setName( 'ga:eventLabel' );
+
+				// Create the ReportRequest object.
+				$profile_id = $analytics->get_data( 'profile-id' );
+				$request    = new Google_Service_AnalyticsReporting_ReportRequest();
+				$request->setViewId( $profile_id );
+				$request->setDateRanges( $date_range );
+				$request->setDimensions(
+					array(
+						$date_dimension,
+						$action_dimension,
+						$label_dimension,
+					)
+				);
+				$request->setMetrics( array( $metrics ) );
+				$request->setDimensionFilterClauses( array( $dimension_filter_clause ) );
+
+				$body = new Google_Service_AnalyticsReporting_GetReportsRequest();
+				$body->setReportRequests( array( $request ) );
+				$ga_options     = new Options( $context );
+				$user_options   = new User_Options( $context );
+				$authentication = new Authentication( $context, $ga_options, $user_options );
+				$client         = $authentication->get_oauth_client()->get_client();
+
+				$analyticsreporting = new Google_Service_AnalyticsReporting( $client );
+				$report_results     = $analyticsreporting->reports->batchGet( $body );
+
+				return $report_results->reports[0]->data->rows;
```

Comment:
will this always be available? might be worth some `isset` checks here if there are cases where no data is returned for this query - for example a brand new analytics account with no data?

Reply:
Good call, done in a6c2d0e3cced3d475964313d6282eaacf58d486c